src/resolver/resolve_path.zig

```diff
@@ -726,111 +707,126 @@ pub const Platform = enum {
-        switch (comptime _platform.resolve()) {
-            .auto => comptime unreachable,
+    pub fn getLastIndexOfSepFunc(comptime platform: Platform) LastSeparatorFunction {
+        comptime switch (platform.resolve()) {
+            .auto => unreachable,
             .loose => {
-                return lastIndexOfSeparatorLoose;
+                return lastIndexOfSepLoose;
             },
             .windows => {
-                return lastIndexOfSeparatorWindows;
+                return lastIndexOfSepWindows;
             },
             .posix => {
-                return lastIndexOfSeparatorPosix;
+                return lastIndexOfSepPosix;
             },
-        }
+        };
     }
 
-    pub inline fn isSeparator(comptime _platform: Platform, char: u8) bool {
-        switch (comptime _platform.resolve()) {
-            .auto => comptime unreachable,
-            .loose => {
-                return isSepAny(char);
+    pub inline fn isSep(comptime platform: Platform, char: u8) bool {
+        switch (comptime platform.resolve()) {
+            .auto => unreachable,
+            inline .posix, .windows => {
+                return std_isSep(char);
             },
-            .windows => {
+            .loose => {
                 return isSepAny(char);
             },
-            .posix => {
-                return isSepPosix(char);
-            },
         }
     }
 
-    pub fn trailingSeparator(comptime _platform: Platform) [2]u8 {
-        return comptime switch (_platform) {
-            .auto => _platform.resolve().trailingSeparator(),
+    pub fn trailingSeparator(comptime platform: Platform) [2]u8 {
+        return comptime switch (platform) {
+            .auto => (platform.resolve()).trailingSeparator(),
             .windows => ".\\".*,
-            .posix, .loose => "./".*,
+            inline .posix, .loose => "./".*,
         };
     }
 
-    pub fn leadingSeparatorIndex(comptime _platform: Platform, path: anytype) ?usize {
-        switch (comptime _platform.resolve()) {
+    pub fn leadingSeparatorIndex(comptime platform: Platform, path: anytype) ?usize {
+        switch (comptime platform.resolve()) {
             .windows => {
+                // Based on isAbsoluteWindowsImpl:
+                // https://github.com/ziglang/zig/blob/993a83081a975464d1201597cf6f4cb7f6735284/lib/std/fs/path.zig#L245
                 if (path.len < 1)
                     return null;
 
-                if (path[0] == '/')
+                if (path[0] == sep_posix)
                     return 0;
 
-                if (path[0] == '\\')
+                if (path[0] == sep_windows)
                     return 0;
 
                 if (path.len < 3)
                     return null;
 
                 // C:\
                 // C:/
-                if (path[0] >= 'A' and path[0] <= 'Z' and path[1] == ':') {
-                    if (path[2] == '/')
```

Comment:
☝️ Remove drive letter check as [zig's standard lib](https://github.com/ziglang/zig/blob/993a83081a975464d1201597cf6f4cb7f6735284/lib/std/fs/path.zig#L245) doesn't do this either.